### PR TITLE
[16.0][FIX] crm_location: make computes multi-record aware

### PR DESCRIPTION
**Pull request by hildickethan**
Adapt computes to be multi-record aware to avoid singleton errors when creating leads in batch

### CODE REVIEW

File: crm_location/models/crm_lead.py [hunk 1]
```diff
@@ -22,20 +22,22 @@ class CrmLead(models.Model):
...
-            self.update(
-                {
-                    "zip": self.location_id.name,
-                    "city": self.location_id.city_id.name,
-                    "state_id": self.location_id.city_id.state_id,
-                    "country_id": self.location_id.city_id.country_id,
-                }
-            )
+        for lead in self:
+            if lead.location_id:
```
Comment (pedrobaeza) [inline crm_location/models/crm_lead.py]: And you avoid a big diff:

```suggestion
        for lead in self.filtered("location_id"):
```

**Review by pedrobaeza — COMMENTED** [review]
And the correct wording would be "multi-record aware compute", not "batch compute", as the computation is not done in a batch way.

**Comment by hildickethan** [comment]
> And the correct wording would be "multi-record aware compute", not "batch compute", as the computation is not done in a batch way.

reworded and added the filtered

**Review by pedrobaeza — APPROVED** [review]
Thanks!

**Review by ioans73 — APPROVED** [review]
LGTM

**Pull request merged**